hs_core/hydroshare/resource.py

```diff
@@ -789,32 +789,8 @@ def delete_resource(pk):
 
     Note:  Only HydroShare administrators will be able to delete formally published resource
     """
-    res = utils.get_resource_by_shortkey(pk)
-    if res.metadata.relations.all().filter(type='isReplacedBy').exists():
-        raise ValidationError('An obsoleted resource in the middle of the obsolescence chain '
-                              'cannot be deleted.')
-
-    # when the most recent version of a resource in an obsolescence chain is deleted, the previous
-    # version in the chain needs to be set as the "active" version by deleting "isReplacedBy"
-    # relation element
-    if res.metadata.relations.all().filter(type='isVersionOf').exists():
-        is_version_of_res_link = \
-            res.metadata.relations.all().filter(type='isVersionOf').first().value
-        idx = is_version_of_res_link.rindex('/')
-        if idx == -1:
-            obsolete_res_id = is_version_of_res_link
-        else:
-            obsolete_res_id = is_version_of_res_link[idx + 1:]
-        obsolete_res = utils.get_resource_by_shortkey(obsolete_res_id)
-        if obsolete_res.metadata.relations.all().filter(type='isReplacedBy').exists():
-            eid = obsolete_res.metadata.relations.all().filter(type='isReplacedBy').first().id
-            obsolete_res.metadata.delete_element('relation', eid)
-            # also make this obsoleted resource editable if not published now that it becomes the latest version
-            if not obsolete_res.raccess.published:
-                obsolete_res.raccess.immutable = False
-                obsolete_res.raccess.save()
-
-    res.delete()
+    from hs_core.tasks import delete_resource_task
+    delete_resource_task(pk, request_username)
```

Comment:
if there isn't a reason to import in the middle of the file we should do it up top

Reply:
Yes, there is a reason which is the circular import issue; otherwise, I'd put it on top for sure. It is captured in this issue: https://github.com/hydroshare/hydroshare/issues/3929 to be addressed holistically later.

Reply:
Can you add a comment in the code linking the issue

Reply:
There are quite a few local imports like this due to circular import issues across our code base, so it may not help much to link the issue here in the comment. I imagine when the issue #3929 is being tackled, a wholistic search to dig out all these local imports is needed to move them up to the top import area.